models.py

```diff
@@ -24,9 +25,12 @@ def __init__(self, node_key):
         self.node_key = node_key
         self.created_at = datetime.utcnow()
 
-    def get_status(self):
-        # TODO: implement status checking
-        return 'active'
+    def is_active(self):
+        last_ping = self.updated_at or self.created_at
+        return last_ping > datetime.utcnow() - NODE_AVAILABILITY_TIMEOUT
+
+    def update_timestamp(self):
```

Comment:
this method can be reused in register_proposal() too

Reply:
Done.